opensrp-immunization/src/main/java/org/smartregister/immunization/util/AppExecutors.java

```diff
@@ -0,0 +1,105 @@
+package org.smartregister.immunization.util;
+
+
+/**
+ * Created by keyman on 12/11/18.
+ */
+
+import android.os.Handler;
+import android.os.Looper;
+import androidx.annotation.NonNull;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * Global executor pools for the whole application.
+ * <p>
+ * Grouping tasks like this avoids the effects of task starvation (e.g. disk reads don't wait behind
+ * webservice requests).
+ */
+public class AppExecutors {
+
+    private static final int THREAD_COUNT = 3;
```

Comment:
Since we already depend on Client Core I think we can just re-use its implementation for this class [here](https://github.com/opensrp/opensrp-client-core/blob/master/opensrp-core/src/main/java/org/smartregister/util/AppExecutors.java)

Reply:
okay. I will reuse the one in client core

Reply:
@ndegwamartin please checkout this [commit](https://github.com/opensrp/opensrp-client-immunization/pull/194/commits/9bc424772e314e60fdef9299b32958f40b0d45e1) and tell me what you think. The reason I had opted to copy over the classes was because of the ease of testing [this constructor](https://github.com/opensrp/opensrp-client-immunization/blob/52282d33bf19b9266ed130bd5e986c32a47e8743/opensrp-immunization/src/main/java/org/smartregister/immunization/util/GenericInteractor.java#L15) provided. I have gone for reflection instead.

Reply:
Yeah I think reflection here is okay for the testing